src/external/listenForDocumentQueries.ts

```diff
@@ -0,0 +1,37 @@
+import { queryApi } from '../client';
+import { FinchQueryOptions } from '../types';
+import { createResponseEvent } from './createEvents';
+import { FinchRequestEvent, FinchDocumentEventNames } from './types';
+
+/**
+ * listenForDocumentQueries binds to document events so websites can
+ * send Finch GraphQL queries and mutations from document events. This
+ * depends if you are able to inject into those pages as well.
+ *
+ * This method will also attach a **data-finch-listener** attribute on
+ * the body element to be able to known the event bound.
+ *
+ * @param options the finch query options that are passed to queryApi method
+ * @returns A function that stops listening to the events
+ */
+export const listenForDocumentQueries = (options?: FinchQueryOptions) => {
+  const onMessage = async (event: FinchRequestEvent) => {
+    const resp = await queryApi(
+      event.detail.query,
+      event.detail.variables,
+      options,
```

Comment:
We should add support to pass the URL here, so then we can also pass that in the message. That way we can add invalidation on the background scripts. Kinda leaning towards just injecting based on the domains we want, but I can see this being a security foot gun if we leave it that way and a dev accidentally adds it into the main content bundle.